plugins/engines/optix/OptiXScene.cpp

```diff
@@ -48,34 +49,116 @@ OptiXScene::OptiXScene(
     optix::Context& context )
     : Scene( renderer, parametersManager )
     , _context( context )
-    , _lightBuffer( 0 )
+    , _lightBuffer( nullptr )
     , _accelerationStructure( DEFAULT_ACCELERATION_STRUCTURE )
-    , _colorMapBuffer( 0 )
-    , _mesh( 0 )
-    , _verticesBuffer( 0 )
-    , _indicesBuffer( 0 )
-    , _normalsBuffer( 0 )
-    , _textureCoordsBuffer( 0 )
-    , _materialsBuffer( 0 )
-    , _volumeBuffer( 0 )
+    , _colorMapBuffer( nullptr )
+    , _mesh( nullptr )
+    , _verticesBuffer( nullptr )
+    , _indicesBuffer( nullptr )
+    , _normalsBuffer( nullptr )
+    , _textureCoordsBuffer( nullptr )
+    , _materialsBuffer( nullptr )
+    , _volumeBuffer( nullptr )
 {
 }
 
 OptiXScene::~OptiXScene()
 {
-    if( _volumeBuffer ) _volumeBuffer->destroy();
-    if( _colorMapBuffer ) _colorMapBuffer->destroy();
+    reset();
+}
+
+void OptiXScene::reset()
+{
+    Scene::reset();
+
+    // Geometry
+    for( auto geometryInstance: _geometryInstances)
+        geometryInstance->destroy();
+    _geometryInstances.clear();
+
+    if( _geometryGroup )
+        _geometryGroup->destroy();
+    _geometryGroup = nullptr;
```

Comment:
This function could be made much simpler by wrapping all the destroyable objects in unique_ptrs with a custom deleter that calls destroy()